Dell/recovery_common.py

```diff
@@ -153,6 +153,45 @@ def check_family(test):
             return True
     return False
 
+def check_install_dhc_id():
+    """The function is used to detect machine's ID for install DHC flow"""
+    path = '/sys/class/dmi/id/product_sku'
+    if not os.path.exists(path):
+        return False
+    with open(path, 'rb') as rfd:
+        value = rfd.read().strip().lower()
+        if not value:
+            return False
+    for top in [ISO_MOUNT, CDROM_MOUNT]:
+        if os.path.isdir(top):
+            plat_conf=os.path.join(top, "DHC", "platform_list", "install-id.conf")
+            if not os.path.exists(plat_conf):
+                return False
+            lines=[line.rstrip('\n') for line in open(plat_conf)]
+            for i in range(len(lines)):
+                if lines[i] in str(value):
+                    return True
+    return False
+    
+def check_recovery_dhc_id():
+    """The function is used to detect machine's ID for recovery DHC flow,"""
+    path = '/sys/class/dmi/id/modalias'
```

Comment:
I'm a bit confused - why isn't this the same as install-id flow?  Don't you want the same stuff offered for recovery or install on the same machines?

Reply:
Because the kernel version currently used for 7070 is 4.15, it does not include product_sku. If want to unify, maybe  all read modalias file